### PR TITLE
#590 - Sentence-level recommender

### PR DESCRIPTION
**Pull request by reckart**
- Basic implementation of a recommender based on the OpenNLP Document Categorizer

### CODE REVIEW

File: inception-imls-opennlp/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/doccat/OpenNlpDoccatRecommenderTest.java [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    public void thatTrainingWorks() throws Exception
+    {
+        OpenNlpDoccatRecommender sut = new OpenNlpDoccatRecommender(recommender, traits);
+        List<CAS> casList = loadDevelopmentData();
+
+        sut.train(context, casList);
+
+        assertThat(context.get(OpenNlpDoccatRecommender.KEY_MODEL))
+            .as("Model has been set")
+            .isNotNull();
```
Comment (Rentier) [inline inception-imls-opennlp/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/doccat/OpenNlpDoccatRecommenderTest.java]: I think that this test does not make sense, as you return now an Optional

File: inception-imls-opennlp/src/test/resources/de/tudarmstadt/ukp/dkpro/core/api/datasets/lib/sentence-classification-en.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+groupId: org.dkpro.core.datasets.sentenceclassification
```
Comment (Rentier) [inline inception-imls-opennlp/src/test/resources/de/tudarmstadt/ukp/dkpro/core/api/datasets/lib/sentence-classification-en.yaml]: Why do we need this file？

Comment (reckart) [reply inception-imls-opennlp/src/test/resources/de/tudarmstadt/ukp/dkpro/core/api/datasets/lib/sentence-classification-en.yaml]: It is used to download the data used in the unit test.

File: inception-imls-opennlp/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/doccat/OpenNlpDoccatRecommenderTest.java (hunk 1 above)
Comment (reckart) [reply inception-imls-opennlp/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/opennlp/doccat/OpenNlpDoccatRecommenderTest.java]: Was copy-pasted from the other test - probably needs to be changed everywhere then.

**Comment by reckart** [comment]
@Rentier good for merge?

**Comment by reckart** [comment]
@Rentier I tested it in the application and it seems to work in general. However, whether it works well or not is a different question (and obviously depends on the task). The accuracy of 0.85 for the argumentative zoning task from the unit test looks nice at a first glance, but there might be a majority class (MISC). Anyway, it is not the purpose of this issue to log more evaluation details or to optimize the results - just to provide the basic functionality.

So, I'll merge it once the checks are good.

**Pull request merged**